ui/src/app/views/workspace-edit/component.ts

```diff
@@ -205,4 +294,50 @@ export class WorkspaceEditComponent implements OnInit {
     return this.accessLevel === WorkspaceAccessLevel.OWNER
         || this.accessLevel === WorkspaceAccessLevel.WRITER;
   }
+
+  keys(input: Object): Array<string> {
+    return Object.keys(input);
+  }
+
+  bucketAsThree(input: Array<string>): Array<Array<string>> {
+    const output: Array<Array<string>> = [];
+    let inputIterator = 0;
+    let currentBucket = -1;
+    input.forEach((inputValue) => {
+      if (inputIterator === 3) {
+        inputIterator = 0;
+      }
+      if (inputIterator === 0) {
+        output.push([]);
+        currentBucket += 1;
+      }
+      output[currentBucket].push(inputValue);
+      inputIterator += 1;
+    });
+    return output;
+  }
+
+  containsUnderserved(enumValue: UnderservedPopulationEnum): boolean {
+    if (this.workspace.researchPurpose.underservedPopulationDetails
+        .find(item => item === enumValue) === undefined) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  switchUnderservedStatus(enumValue: UnderservedPopulationEnum): void {
+    if (this.mode === WorkspaceEditMode.Edit) {
+      return;
+    }
+    const positionOfValue = this.workspace.researchPurpose.
```

Comment:
This would be easier to read with some intermediary variables.

```
const details = this./* long path to the array in question */;
const index = details.indexOf(enumValue);
if (index > -1) {
  details.splice(index, 1);
} else {
  details.push(enumValue);
}
```

JS is (sort of) copy by ref, not value, so you shouldn't need to reassign `details` to the long path.  But if you do, its still worth it for the gain in brevity and less line wrapping.